mlflow/tensorflow.py

```diff
@@ -989,110 +985,118 @@ def _log_early_stop_callback_metrics(callback, history, metrics_logger):
-                # Checking if the 'callback' argument of fit() is set
-                if len(args) >= 6:
-                    tmp_list = list(args)
-                    early_stop_callback = _early_stop_check(tmp_list[5])
-                    tmp_list[5], log_dir = _setup_callbacks(tmp_list[5], log_models, metrics_logger)
-                    args = tuple(tmp_list)
-                elif kwargs.get("callbacks"):
-                    early_stop_callback = _early_stop_check(kwargs["callbacks"])
-                    kwargs["callbacks"], log_dir = _setup_callbacks(
-                        kwargs["callbacks"], log_models, metrics_logger
-                    )
-                else:
-                    kwargs["callbacks"], log_dir = _setup_callbacks([], log_models, metrics_logger)
+    def fit(original, self, *args, **kwargs):
+        active_run = mlflow.active_run()
+        if MLFLOW_AUTOLOGGING in active_run.data.tags:
+            global _AUTOLOG_RUN_ID
+            _AUTOLOG_RUN_ID = active_run.info.run_id
+
+        unlogged_params = ["self", "x", "y", "callbacks", "validation_data", "verbose"]
+
+        log_fn_args_as_params(original, args, kwargs, unlogged_params)
+        early_stop_callback = None
+
+        run_id = mlflow.active_run().info.run_id
+        _AUTOLOG_RUN_ID = run_id
+
+        with batch_metrics_logger(run_id) as metrics_logger:
+            # Checking if the 'callback' argument of fit() is set
+            if len(args) >= 6:
+                tmp_list = list(args)
+                early_stop_callback = _early_stop_check(tmp_list[5])
+                tmp_list[5], log_dir = _setup_callbacks(tmp_list[5], log_models, metrics_logger)
+                args = tuple(tmp_list)
+            elif kwargs.get("callbacks"):
+                early_stop_callback = _early_stop_check(kwargs["callbacks"])
+                kwargs["callbacks"], log_dir = _setup_callbacks(
+                    kwargs["callbacks"], log_models, metrics_logger
+                )
+            else:
+                kwargs["callbacks"], log_dir = _setup_callbacks([], log_models, metrics_logger)
 
-                _log_early_stop_callback_params(early_stop_callback)
+            _log_early_stop_callback_params(early_stop_callback)
 
-                history = original(self, *args, **kwargs)
+            history = original(self, *args, **kwargs)
 
-                _log_early_stop_callback_metrics(early_stop_callback, history, metrics_logger)
+            _log_early_stop_callback_metrics(early_stop_callback, history, metrics_logger)
 
-            _flush_queue()
-            _log_artifacts_with_warning(
-                local_dir=log_dir.location, artifact_path="tensorboard_logs"
-            )
-            if log_dir.is_temp:
-                shutil.rmtree(log_dir.location)
+        _flush_queue()
+        _log_artifacts_with_warning(local_dir=log_dir.location, artifact_path="tensorboard_logs")
+        if log_dir.is_temp:
+            shutil.rmtree(log_dir.location)
```

Comment:
can we leverage `mlflow.utils.autologging_utils.PatchFunction.on_exception` here to ensure that the logdir is removed if an unexpected exception occurs? see https://github.com/dbczumar/mlflow/blob/da54e042ef52c4a5c4314bf320bcb036ebdf2b1e/mlflow/tensorflow.py#L992-L1045 for an example.

Reply:
done!